wgpu-core/src/device/mod.rs

```diff
@@ -1167,7 +1167,7 @@ impl<B: GfxBackend> Device<B> {
                             if read_only {
                                 resource::BufferUse::STORAGE_LOAD
                             } else {
-                                resource::BufferUse::STORAGE_STORE
+                                resource::BufferUse::STORAGE_LOAD | resource::BufferUse::STORAGE_STORE
```

Comment:
I believe that would make us throw a validation error because of conflicting usage.
The fix needs to be at a lower level.